src/script/engines/lua.c

```diff
@@ -976,11 +1051,19 @@ bool _luaLoad(struct mScriptEngineContext* ctx, const char* filename, struct VFi
 		// Make the old _ENV the __index in the metatable
 		lua_newtable(luaContext->lua);
 		lua_pushliteral(luaContext->lua, "__index");
+#if LUA_VERSION_NUM >= 502
 		lua_getupvalue(luaContext->lua, -4, 1);
+#else
+		lua_getfenv(luaContext->lua, -4);
+#endif
```

Comment:
lua_getupvalue exists with the same signature in 5.1. What's different in 5.1 that this is needed?

Reply:
This _is_ the breaking change between 5.1 and 5.2+ -- in 5.2+ the first upvalue of a function is the lexical environment (if it exists -- you cannot assume a function inherits a lexical environment), but in 5.1 the lexical environment is managed by getfenv/setfenv.